### PR TITLE
Correctly handle the DescSet relocation

### PR DESCRIPTION
**Pull request by s-perron**
Code was added to generate a relocation for the off of the descriptor
table for a given descriptor set.  However, the code the replace the
relocation was not written.  I cannot remember why, but it was probably
because I could not create a test that actaully needed it.  The code
added and forgotten.  I have not found a case that will generate it.
I'm adding a test and adding the code to replace the relocation.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 388957caf85591b72c96b459d0632f89d97b0b3c
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev f49a7f3f706dd721d9a7b620be1482217ef63b6e</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev c3d047085243c811583a612b5ae016a513705465</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 96c47aa27a6b84f17c088126f485d2c9d7569a22
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev f49a7f3f706dd721d9a7b620be1482217ef63b6e</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev c3d047085243c811583a612b5ae016a513705465</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2096a7f8528dda54bf84731268c960017dbdd385
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev f49a7f3f706dd721d9a7b620be1482217ef63b6e</li>
<li>SPVGEN: dev efa3cec20c6ebc66254ff9cca9b3ea124811d7a3</li>
<li>XGL: dev c3d047085243c811583a612b5ae016a513705465</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225388)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31425/56347 (55.8%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24922/56347 (44.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31413/56347 (55.7%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24934/56347 (44.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31748/56347 (56.3%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 24599/56347 (43.7%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34582/56347 (61.4%)</li>
<li>Failed: 0/56347 (0.0%)</li>
<li>Not Supported: 21765/56347 (38.6%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit eb11b77861c7a6e9821899729a4c0e99cd8610d3
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/163959)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li>
</ul></details>


</ul></details>

File: lgc/include/lgc/state/AbiUnlinked.h [hunk 1]
```diff
@@ -63,6 +63,16 @@ const static char DescriptorOffset[] = "doff_";
 //  * 1: descriptor pointer should be loaded from the spill table
 const static char DescriptorUseSpillTable[] = "dusespill_";
 
+// Whether the descriptor pointer comes from a spill table.
+// The format is: "dusespill_X_Y" where:
+// * X is the descriptor set number
+// * Y is the binding number
+//
```
Comment (kuhar) [inline lgc/include/lgc/state/AbiUnlinked.h:70]: This seems to be duplicated from above the previous string.

**Review by kuhar — COMMENTED** [review]
I don't see anything wrong with the logic but I'm not familiar with this code. It'd be better if @trenouf took a look too.

File: lgc/include/lgc/state/AbiUnlinked.h (hunk 1 above)
Comment (s-perron) [reply lgc/include/lgc/state/AbiUnlinked.h]: Right, I forgot to fix that up.

**Comment by s-perron** [comment]
V2: Fixed the comment describing the relocation.

**Review by trenouf — APPROVED** [review]
LGTM

**Comment by trenouf** [comment]
We should wait for the jenkins build before merging.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 90c4f965346cc9fc9c678a9971740c20f7dd72c8
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 44abc48d199dd28369bf7415eee1751c4bf616d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev b8ed024d536d1bebb35927a0f4f470dca2f2dfc6</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev cdd74831d061bf65fdf5af4a0b6ce99fe2c558a9</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/218612)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32354/54653 (59.2%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22299/54653 (40.8%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32194/54653 (58.9%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22458/54653 (41.1%)</li><li>Warnings: 1/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32530/54653 (59.5%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 22123/54653 (40.5%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35511/54653 (65.0%)</li>
<li>Failed: 0/54653 (0.0%)</li>
<li>Not Supported: 19142/54653 (35.0%)</li><li>Warnings: 0/54653 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**